File: src/Q5/Solution.java
package Q5;

public class Solution {
    public static String longestPalindrome(String s) {
        for (int j = s.length(); j >= 0; j--) {
            for (int i = 0; i <= s.length() - j; i++) {
                if (isPalindrome(s.substring(i, j + i))) {
                    return s.substring(i, j + i);
                }
            }
        }
        return "";
    }

    public static boolean isPalindrome(String s) {
        for (int i = 0; i < s.length() / 2; i++) {
            if (s.charAt(i) != s.charAt(s.length() - i - 1)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
//        isPalindrome("Hello");
        System.out.println(longestPalindrome("abcdefghijk"));
        System.out.println(longestPalindrome("cbbd"));
    }
}

File: src/Q5/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        # Sliding Window
        longest = 1
        length = len(s)
        left_i = 0
        right_i = 1

        for i in range(len(s) - 1):
            for j in range(2):
                left = i
                right = i + j
                while left >= 0 and right < length and s[left] == s[right]:
                    if right + 1 - left > longest:
                        longest = right - left + 1
                        left_i = left
                        right_i = right + 1
                    left -= 1
                    right += 1
        return s[left_i:right_i]


if __name__ == '__main__':
    s = Solution()
    print(s.longestPalindrome("babad"))
    print(s.longestPalindrome("cbbd"))
